Sources/PackageModel/Product.swift

```diff
@@ -70,3 +70,18 @@ extension Product: CustomStringConvertible {
         }
     }
 }
+
+extension ProductType: Equatable {}
+public func ==(lhs: ProductType, rhs: ProductType) -> Bool {
+    switch (lhs, rhs) {
+    case (.Executable, .Executable):
+        return true
+    case (.Library(.Static), .Library(.Static)):
+        return true
+    case (.Library(.Dynamic), .Library(.Dynamic)):
+        return true
+    case (.Test, .Test):
+        return true
+    default: return false
+    }
```

Comment:
BTW, one trick for writing these functions is:
```
switch (lhs, hrs) {
case (.Exectuable, .Executable):
  return true
case (.Executable, _):
  return false
...
}
```
which is longer, but lets you avoid the `default` case which means that this code will fail if it isn't updated when the enum is extended.